internal/db/migrations/002_utility_functions.up.sql

```diff
@@ -28,4 +46,14 @@ BEGIN
 END ;
 $reset_sequences$ LANGUAGE 'plpgsql';
 
+-- Resource tags must never be changed, this function allows triggers to enforce it
+CREATE OR REPLACE FUNCTION prevent_tag_update()
+  RETURNS trigger AS
+$prevent_tag_update$
+BEGIN
+  NEW.tag := OLD.tag;
+  RETURN NEW;
+END;
+$prevent_tag_update$ LANGUAGE 'plpgsql';
+
 COMMIT;
```

Comment:
Another bunch of helper functions, database must be recreated in order to re-migrate it. Sorry, I do not want to do this in separate migrations to keep migration history clean.